pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -186,11 +216,11 @@ protected BrokerResponse handleRequest(long requestId, String query, SqlNodeAndO
       return new BrokerResponseNative(QueryException.getException(QueryException.QUERY_VALIDATION_ERROR, e));
     } catch (WebApplicationException e) {
       throw e;
-    } catch (RuntimeException e) {
-      String consolidatedMessage = ExceptionUtils.consolidateExceptionMessages(e);
+    } catch (Throwable t) {
+      String consolidatedMessage = ExceptionUtils.consolidateExceptionMessages(t);
       LOGGER.warn("Caught exception planning request {}: {}, {}", requestId, query, consolidatedMessage);
       _brokerMetrics.addMeteredGlobalValue(BrokerMeter.REQUEST_COMPILATION_EXCEPTIONS, 1);
-      if (e.getMessage().matches(".* Column .* not found in any table'")) {
+      if (t.getMessage().matches(".* Column .* not found in any table'")) {
```

Comment:
just an error I've found: t.getMessage may be null. Given you are touching this line, can you also modify this line to check if the message is null?